### PR TITLE
Cairo lang 0.8.2

### PR DESCRIPTION
**Pull request by FabijanC**
## Usage related changes
- Adapt to cairo-lang 0.8.2
- Default to the latest block in getBlock
- Update to devnet 0.2.1

## Development related changes
- Introduce type BlockNumber that can be `number`, `"pending"`, or `"latest"`

TODO:
- Should defaulting to `pending` be replaced with defaulting to `latest`?
- Where else is defaulting not implemented?

# Checklist:

- [x] I have formatted the code
- [x] I have performed a self-review of the code
- [x] I have rebased to the base branch
- [x] I have documented the changes
- [x] I have updated the tests
- [x] I have created a PR to the `plugin` branch of [`starknet-hardhat-example`](https://github.com/Shard-Labs/starknet-hardhat-example).

### CODE REVIEW

File: src/types.ts [hunk 1]
```diff
@@ -598,6 +599,7 @@ export class StarknetContract {
         args?: StringMap,
         options: EstimateFeeOptions = {}
     ): Promise<FeeEstimation> {
+        options = copyWithBigint(options); // copy because of potential changes to the object
```
Comment (badurinantun) [inline src/types.ts:602]: If have to add comments that means this code is hard to understand by itself. We shouldn't change the function arguments, but create a new object with the correct block number

File: config.json [hunk 2]
```diff
@@ -1,4 +1,4 @@
 {
-    "CAIRO_LANG": "0.8.1",
-    "STARKNET_DEVNET": "0.2.0"
+    "CAIRO_LANG": "0.8.2",
```
Comment (badurinantun) [inline config.json:2]: What about `0.8.2.1`?

**Review by badurinantun — CHANGES_REQUESTED** [review]

File: config.json (hunk 2 above)
Comment (FabijanC) [reply config.json:2]: totally forgot about that

File: src/types.ts (hunk 1 above)
Comment (FabijanC) [reply src/types.ts:602]: this is actually old code, just noticed that it was missing the copying part; but I can try to edit it

File: config.json (hunk 2 above)
Comment (FabijanC) [reply config.json]: https://github.com/Shard-Labs/starknet-hardhat-plugin/pull/111/commits/e9cbce03eb3528eee7c8d7d7ed6ae01a40385215

File: src/types.ts (hunk 1 above)
Comment (FabijanC) [reply src/types.ts]: https://github.com/Shard-Labs/starknet-hardhat-plugin/pull/111/commits/4b20ead70affda21c4b2ba9f2ed6296ea33bdc01

**Review by badurinantun — APPROVED** [review]

**Pull request merged**